src/Main/Main.module.scss

```diff
@@ -4,9 +4,15 @@ $back-to-top-height: rem(
   $nds-spacing-medium * 3
 ); // Line height + bottom and top padding
 
+$main-content-padding: 4rem;
+
 @supports (position: sticky) {
   .main {
     padding-bottom: $back-to-top-height; /* space for the back to top link to dock */
     position: relative;
+
+    &.withPadding {
+      padding-bottom: $main-content-padding; /* space between page content and footer when withPadding prop set to true */
```

Comment:
I wonder if this makes more sense to be `$back-to-top-height + rem($nds-spacing-large)`? As in the the `$main-content-padding` variable isn't clear if it's the padding on the main element or the gap above the back to top button, so doing the addition here makes this clearer